src/gurt/telemetry.c

```diff
@@ -2522,15 +2586,18 @@ d_tm_add_ephemeral_dir(struct d_tm_node_t **node, size_t size_bytes,
 		D_ERROR("can't set up the link node, " DF_RC "\n", DP_RC(rc));
 		D_GOTO(fail_tracking, rc);
 	}
-	D_ASSERT(link_node->dtn_type == D_TM_LINK);
-	link_node->dtn_metric->dtm_data.value = key;
 
 	/* track attached regions within the parent shmem */
 	parent_shmem = get_shmem_for_key(ctx, link_node->dtn_shmem_key);
 	if (parent_shmem == NULL) {
 		D_ERROR("failed to get parent shmem pointer\n");
 		D_GOTO(fail_link, rc = -DER_NO_SHMEM);
 	}
+
+	D_ASSERT(link_node->dtn_type == D_TM_LINK);
```

Comment:
I'm tripping this assert on Frontera. See this comment for details
https://daosio.atlassian.net/browse/DAOS-15083?focusedCommentId=124040

Reply:
I can no longer reproduce this, so maybe something strange in my environment. I'm running performance testing now

Reply:
@daltonbohning Any updates on the test results?

Reply:
Funny. I _**just**_ pulled them together :)
I ran IOR Easy with SX on Frontera, based on our standard performance suite. There doesn't seem to be any significant difference between `D_CLIENT_METRICS_ENABLE=0`, `D_CLIENT_METRICS_ENABLE=1`, and master.

Unless we think more performance testing is needed, I think it's okay to move forward.

Reply:
After discussing this internally, we need to check the performance overhead with many RPCs. The IOR Easy tests I ran use 1M transfer, so there aren't many RPCs. I'll verify with MDTest Hard, which has many smaller IOs

Reply:
The queue has been busy, so most of my performance jobs are still pending